### PR TITLE
Reland: "Update outdated runners in the benchmarks folder (#91126)"

### PR DESCRIPTION
**Pull request by gspencergoog**
## Description

This reverts commit b412f6fff89e161a2d56e1b4cfc572d7eede2c3a (#92106), and fixes the build issues for Android builds.

original description:
> This PR updates the platform runners in the benchmarks subdir.

> I've attempted to replace the entire group with new code as if the project were recreated with `flutter create` so that they don't become a Frankenstein of minor changes to make it work again, but just plain old generated code that can be replaced and updated in place.

## Tests
 - No tests because we're just updating boilerplate here.

### CODE REVIEW

**Comment by blasten** [comment]
There are some conflicts unfortunately

**Comment by gspencergoog** [comment]
@blasten, could you take another look, I've resolved the conflicts.

**Comment by skia-gold** [comment]
Gold has detected about 20 new digest(s) on patchset 4.
View them at https://flutter-gold.skia.org/cl/github/92535

**Review by blasten — APPROVED** [review]
LGTM

File: dev/benchmarks/complex_layout/android/app/build.gradle [hunk 1]
```diff
@@ -15,22 +15,43 @@ if (flutterRoot == null) {
...
+apply plugin: 'kotlin-android'
 apply from: "$flutterRoot/packages/flutter_tools/gradle/flutter.gradle"
 
 android {
     compileSdkVersion flutter.compileSdkVersion
 
     compileOptions {
-        sourceCompatibility JavaVersion.VERSION_11
-        targetCompatibility JavaVersion.VERSION_11
+        sourceCompatibility JavaVersion.VERSION_1_8
```
Comment (blasten) [inline dev/benchmarks/complex_layout/android/app/build.gradle:36]: this fixes https://github.com/flutter/flutter/issues/92859

**Pull request merged**